packages/framework/fluid-static/src/utils.ts

```diff
@@ -51,10 +51,11 @@ export function isDataObjectClass(obj: unknown): obj is InternalDataObjectClass<
 /**
  * Runtime check to determine if a class is a SharedObject type
  */
-export const isSharedObjectClass = (obj: unknown): obj is SharedObjectClass<IFluidLoadable> => {
-	const maybe = obj as Partial<SharedObjectClass<IFluidLoadable>> | undefined;
-	return maybe?.getFactory !== undefined;
-};
+export function isSharedObjectKind(
+	obj: LoadableObjectClass,
+): obj is ISharedObjectKind<IFluidLoadable> {
+	return !isDataObjectClass(obj);
```

Comment:
this is a non package exported method which sorts between ISharedObjectKind and DataObjectClass. Since the input is required to be one or the other, and the typescript type system ignores the ability of a type to be both of them at once, I think its fine for our runtime behavior to match that.

Also I think its more likely for a DataObjectClass to happen to have a member named `getFactory` than for a ISharedObjectKind to happen to have a member named `factory` with a member named `IFluidDataStoreFactory` so I think this is actually more robust than the old check (less likely to give a false positive with a new negligible risk of a false negative)

Reply:
Dropping a runtime check to rely on a type check assumes the customer is using Typescript and not plain JS as a means to prevent them from passing something more garbage in.  I would not recommend that assumption for the broader customer base.

Not saying this check is/was great, but since changing it doesn't seem core to the point of this PR maybe prefer status quo.  Not blocking, up to your judgement.

Reply:
I have added this check to isDataObjectClass and used it to validate the input, a well as added a TODO since this down casting isn't fully safe and should be possible to fix.